flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/TranslationContext.scala

```diff
@@ -59,29 +64,55 @@ object TranslationContext {
       .traitDefs(ConventionTraitDef.INSTANCE)
       .build
 
-    tabNames = Map[AbstractTable, String]()
-
+    tablesRegistry = Map[String, AbstractTable]()
     relBuilder = RelBuilder.create(frameworkConfig)
-
     nameCntr.set(0)
 
   }
 
   def addDataSet(newTable: DataSetTable[_]): String = {
+    val tabName = "DataSetTable_" + nameCntr.getAndIncrement()
+    tables.add(tabName, newTable)
+    tabName
+  }
+
+  @throws[TableException]
+  def addAndRegisterDataSet(table: DataSetTable[_], name: String): Unit = {
```

Comment:
I think `addAndRegisterDataSet(DataSetTable, String)` and `registerTable(TableTable, String)` can be combined to `registerTable(AbstractTable, String)`